### ISSUE TITLE
Casting a float to a string sometimes results in a string, not a numeric-string

### ISSUE THREAD

**Issue by BOPOH** [vimeo/psalm#10051]
https://psalm.dev/r/21739e19e1 - no issues but it fails in PHP. I think it should be catched and produces an error or a warning

**Comment by BOPOH** [comment]
sorry, it seems like a bug in the bcmath:
```php
$float = 1e-10;
var_dump([
    (string)$float,        // "1.0E-10"
    (float)(string)$float, // double(1.0E-10)
]);
```
ie string "1.0E-10" is still a correct numeric string
